optuna/samplers/_cmaes.py

```diff
@@ -433,7 +434,7 @@ def sample_relative(
         else:
             params = optimizer.ask()
 
-        generation_attr_key = self._attr_keys.generation
+        generation_attr_key = self._attr_prefix + "generation"
```

Comment:
Nit: It might be better to introduce the following two property methods to avoid inconsistencies in the key definitions.

```python
@property
def _attr_key_generation(self) -> str:
    return self._attr_prefix + "generation"

@property
def _attr_key_optimizer(self, index: int) -> str:
    return f"{self._attr_prefix}optimizer:{index}"
```